internal/server/event_server_test.go

```diff
@@ -20,29 +20,39 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"testing"
 	"time"
 
-	"github.com/onsi/gomega"
+	. "github.com/onsi/gomega"
 	"github.com/sethvargo/go-limiter/httplimit"
 	"github.com/sethvargo/go-limiter/memorystore"
+	"github.com/sethvargo/go-limiter/noopstore"
+	"github.com/slok/go-http-metrics/middleware"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/kubectl/pkg/scheme"
```

Comment:
We should be able to get rid of this new dependency by using `k8s.io/apimachinery/pkg/runtime` to create a new scheme. Refer https://github.com/fluxcd/notification-controller/blob/610fd63c11872134deb65f69c6bfaa5f00412bbd/internal/server/receiver_handler_test.go#L272